controllers/ovndbcluster_controller.go

```diff
@@ -749,6 +751,12 @@ func (r *OVNDBClusterReconciler) reconcileServices(
 		if err != nil {
 			return ctrl.Result{}, err
 		}
+		// It can be possible that not all pods are ready, so DNSData won't
+		// have complete information, return error to retrigger reconcile loop
+		if len(podList.Items) < int(*(instance.Spec.Replicas)) {
+			err = fmt.Errorf("error not all pods are yet created, number of expected pods: %v, current pods: %v", *(instance.Spec.Replicas), len(podList.Items))
```

Comment:
"error" in the message is not needed. (it will be logged as error level I believe.)

`fmt.Errorf('some pods are not created yet (%d expected, %d actual)')`